### PR TITLE
feat: ADDRESS opcode

### LINKED ISSUE #30: Opcode: ADDRESS

### PR DESCRIPTION
**Pull request by ricomateo**
Closes #30 
This PR implements code generation for the [ADDRESS](https://www.evm.codes/#30?fork=cancun) opcode.

### CODE REVIEW

File: tests/evm.rs [hunk 1]
```diff
@@ -24,6 +28,7 @@ fn run_program_assert_result(
     let mut evm = Evm::new(env, program);
     let result = evm.transact();
     assert!(&result.is_success());
+    println!("bytes = {:?}", result.return_data().unwrap());
```
Comment (JulianVentura) [inline tests/evm.rs:31]: I think you forgot this log here.

File: tests/evm.rs [hunk 2]
```diff
@@ -454,3 +459,33 @@ fn callvalue_stack_overflow() {
     let env = Env::default();
     run_program_assert_halt(program, env);
 }
+
+#[test]
+fn address() {
```
Comment (JulianVentura) [inline tests/evm.rs:464]: Can we add a test (or change this one if you want) to match the [EVM Codes example for ADDRESS](https://www.evm.codes/playground?unit=Wei&codeType=Mnemonic&code=%27ADDRESS%27_), please?

File: src/env.rs [hunk 3]
```diff
@@ -152,3 +152,13 @@ pub enum TransactTo {
     /// Contract creation.
     Create,
 }
+
+impl TxEnv {
+    pub fn get_address(&self) -> [u8; 20] {
```
Comment (JulianVentura) [inline src/env.rs:157]: nit: If the function is `get_address`, then I would return an `Address` struct here. Then if you want to convert that struct to bytes you can do that in the caller function.

**Comment by JulianVentura** [comment]
Great! LGTM

File: tests/evm.rs [hunk 4]
```diff
@@ -454,3 +479,37 @@ fn callvalue_stack_overflow() {
...
+    let operations = vec![Operation::Address];
+    let mut env = Env::default();
+    env.tx.transact_to = TransactTo::Call(address);
+    let address_bytes = &address.to_fixed_bytes();
+    //We extend the result to be 32 bytes long.
+    let expected_result: [u8; 32] = [&[0u8; 12], &address_bytes[0..20]]
+        .concat()
+        .try_into()
+        .unwrap();
+    run_program_assert_bytes_result(operations, env, &expected_result);
```
Comment (MegaRedHand) [inline tests/evm.rs:495]: Can't we use `run_program_assert_result` with `BigUint::from_be_bytes(address_bytes)` ?

**Review by MegaRedHand — APPROVED** [review]
LGTM

**Pull request merged**